docs2/site/docs/getting-started/custom-scalars.md

```diff
@@ -124,51 +132,215 @@ public void ConfigureServices(IServiceCollection services)
+only need to override this method if you are creating a custom scalar that returns enumeration
+values, or if you are returning structured data.
 
-public class ExampleSchema : Schema
+### 8. Code-first: Register the .NET type mapping with the schema (optional).
+
+In order to allow simple field mappings like `Field(x => x.Vector)`, you need to register the
+custom scalar with the schema so that the proper graph type can be selected for the field.
+Call `RegisterTypeMapping` with the CLR data model type and the graph type you want it mapped to.
+
+```csharp
+public class MySchema
 {
-    public ExampleSchema(IServiceProvider provider)
-        : base(provider)
-    {        
-        ValueConverter.Register<string, Vector3>(ParseVector3);
+    public MySchema()
+    {
+        ...
+
+        this.RegisterTypeMapping<Vector3, Vector3Type>();
+    }
+}
+```
+
+This is not necessary if you use the alternate `Field<T>` syntax which specifies the graph type
+to be used for the field.
+
+In this example, you created a custom scalar. In summary:
+
+- Create a data class for the server-side representation of the scalar
+- Implement a `ScalarGraphType` which handles variable parsing, literal parsing, and serialization
+- Register the `ScalarGraphType` within the DI container
+
+You can also choose to override `CanParseLiteral`, `CanParseValue` or `IsValidDefault` for
+enhanced performance. The default implementations call `ParseLiteral`, `ParseValue` and
+`ToAST` respectively, returning `false` if an exception is caught, or `true` otherwise.
+If you do choose to implement these methods, note that those methods must not throw an
+exception, and that they are not always called when executing a document.
 
-        // Other schema assignments...
+## Vector3 sample with combined string/structured parsing and serialization
+
+Keep in mind that the serialized value returned by custom scalar can by anything that the
+environment allows. For example it can be a structured object, rather than a simple value.
+This is also true of variables, however literals can only be the represented by the native
+scalar types supported by the GraphQL query language -- integers, floating-point values,
+booleans and null values. You cannot parse a structured literal object into a scalar value.
+
+So to extend our sample, let's assume that we want the Vector3 scalar to instead accept and
+return data in a more structured format, in addition to supporting the string format for literals.
+
+Here is a sample of a variable supporting a more structured format:
+
+```graphql
+mutation AddVector($vector3: Vector3!) {
+    addVector(vector3: $vector3)
+}
+
+//variables
+{
+    "vector3": {
+        "X":"23",
+        "Y":"43",
+        "Z":"66"
     }
+}
+```
 
-    private Vector3 ParseVector3(string vector3InputString)
+And a sample of a response with a vector in a more structured format:
+
+```json
+{
+    "data": {
+        "getVector": {
+            "X":"23",
```

Comment:
By the way... should we change "X" to "x" in all requests/responses?

Reply:
Yes